components/build/tekton-chains/chains-certs.yaml

```diff
@@ -113,3 +113,38 @@ spec:
       terminationGracePeriodSeconds: 30
       serviceAccount: chains-secrets-admin
       serviceAccountName: chains-secrets-admin
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: tekton-chains-signing-secret
+  namespace: tekton-chains
+  annotations:
+    argocd.argoproj.io/hook: Sync
+    argocd.argoproj.io/hook-delete-policy: HookSucceeded
+spec:
+  template:
+    spec:
+      containers:
+      - name: chains-secret-generation
+        image: quay.io/redhat-appstudio/appstudio-utils:eb94f28fe2d7c182f15e659d0fdb66f87b0b3b6b
+        imagePullPolicy: Always
+        command:
+        - /bin/bash
+        - -c
+        - |
+          # Once the key-pair has been set it's marked as immutable so it can't be updated.
+          # Try to handle that nicely. The object is expected to always exist so check the data.
+          SIG_KEY_DATA=$(kubectl get secret signing-secrets -n tekton-chains -o jsonpath='{.data}')
+          [[ -n $SIG_KEY_DATA ]] && echo "Signing secret exists." && exit
+
+          # To make this run conveniently without user input let's create a random password
+          RANDOM_PASS=$( head -c 12 /dev/urandom | base64 )
+
+          # Generates the key pair secret directly in the cluster. Also creates public key file cosign.pub
```

Comment:
Maybe no need to mention the public key file in this comment, since we aren't going to save it.